packages/zowe-explorer/src/extension.ts

```diff
@@ -60,11 +62,101 @@ const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+        // Migrate old settings to new settings if any old settings exist and migration has not been run yet
+        if (zoweOldConfigurations.length > 0) {
+            zoweOldConfigurations.forEach((configuration) => {
+                // Retrieve the old setting for both scopes
+                let globalValue: any = globalIsNotMigrated
+                    ? configurations.inspect(configuration).globalValue
+                    : undefined;
+                let workspaceValue: any = workspaceIsNotMigrated
+                    ? configurations.inspect(configuration).workspaceValue
+                    : undefined;
+
+                // Handle edge case where Zowe-Temp-Folder-Location is migrated but has new schema
+                // Reassign value only if object retrieved is not undefined
+                if (configuration === "Zowe-Temp-Folder-Location") {
+                    globalValue = globalValue ? globalValue.folderPath : globalValue;
+                    workspaceValue = workspaceValue ? workspaceValue.folderPath : workspaceValue;
+                }
+
+                const newSetting = configurationDictionary[configuration];
+
+                // Handle case where a configuration could be in either workspace or global settings and determine where to migrate
+                if (globalIsNotMigrated && globalValue !== undefined && workspaceValue === undefined) {
+                    configurations.update(newSetting, globalValue, vscode.ConfigurationTarget.Global);
+                    globalIsNotMigrated = true;
+                } else if (workspaceIsNotMigrated && workspaceValue !== undefined && globalValue === undefined) {
+                    configurations.update(newSetting, workspaceValue, vscode.ConfigurationTarget.Workspace);
+                    workspaceIsNotMigrated = true;
+                } else if (
+                    workspaceIsNotMigrated &&
+                    globalIsNotMigrated &&
+                    workspaceValue !== undefined &&
+                    globalValue !== undefined
+                ) {
+                    configurations.update(newSetting, globalValue, vscode.ConfigurationTarget.Global);
+                    configurations.update(newSetting, workspaceValue, vscode.ConfigurationTarget.Workspace);
+                }
+            });
+
+            // Confirm migration is completed so it will not run more than once for either global or workspace settings
+            if (globalIsNotMigrated) {
+                configurations.update(
+                    globals.SETTINGS_VERSION,
+                    currentVersionNumber,
+                    vscode.ConfigurationTarget.Global
+                );
+            }
+            if (workspaceIsNotMigrated) {
+                configurations.update(
+                    globals.SETTINGS_VERSION,
+                    currentVersionNumber,
+                    vscode.ConfigurationTarget.Workspace
+                );
+            }
+        }
+    }
+
```

Comment:
@rudyflores to keep activate() simpler to read and understand should we move this new section of code into a new file in the utils folder and call it from activate like checkConfigurations() or something?  @lauren-li what do you think?

Reply:
@JillieBeanSim @rudyflores Yes, I agree it should at least be a separate function! (Or maybe two functions? One for checking configuration version and one for doing the actual migration.) I think we had originally meant to separate it out from `activate()` once it was all working, but we got too excited once all the tests were finally passing and it slipped our minds.

Reply:
I like this idea, this would help with maintaining the code :)